openedxstats/apps/sites/views.py

```diff
@@ -81,13 +81,14 @@ class HawthornMapView(generic.ListView):
     # context_object_name = 'sites_map'
 
 def stats_view(request):
-    active_sites_count = Site.objects.exclude(active_end_date__isnull=False).count()
-    active_sites_course_count = Site.objects.exclude(active_end_date__isnull=False).aggregate(Sum('course_count'))['course_count__sum']
+    active_sites = Site.objects.exclude(active_end_date__isnull=False).filter((Q(course_count__gt=0) | Q(is_private_instance=True)) & Q(is_gone=False)).aggregate(sites=Count('*'), courses=Sum('course_count'))
```

Comment:
Line 196 has:
```
            day_stats = Site.objects.filter(
                (Q(course_count__gt=0) | Q(is_private_instance=True)) &
                Q(is_gone=False) &
                date_select
                ).etc
 ```
so it seems like we could create a function to handle the core of it.